src/wazuh_modules/vulnerability_scanner/src/scanOrchestrator/eventDetailsBuilder.hpp

```diff
@@ -229,6 +231,23 @@ class TEventDetailsBuilder final : public AbstractHandler<std::shared_ptr<TScanC
                 auto vulnerabilityDetection = PolicyManager::instance().getVulnerabilityDetection();
                 ecsData["wazuh"]["cluster"]["name"] = data->clusterName();
                 ecsData["wazuh"]["schema"]["version"] = WAZUH_SCHEMA_VERSION;
+                switch (data->affectedComponentType())
+                {
+                    case AffectedComponentType::Package:
+                        ecsData["wazuh"]["vulnerability"]["source"] = TGlobalData::instance().adpDescriptions().at(
+                            m_databaseFeedManager->getCNA(data->packageFormat().data(),
```

Comment:
Question: could we make the code safer here handling the case where the new `base-rules/adp-global.json` doesn't contain a recently added vendor to `base-rules/feed-global.json` ?

Reply:
Yeap, I'm working on the UTs and I'll add something about that